core/server/models/api-key.js

```diff
@@ -52,7 +57,7 @@ const ApiKey = ghostBookshelf.Model.extend({
         // - content key = no role
         if (this.hasChanged('type') || this.hasChanged('role_id')) {
             if (this.get('type') === 'admin') {
-                return Role.findOne({name: 'Admin Integration'}, Object.assign({}, options, {columns: ['id']}))
+                return Role.findOne({name: attrs.role || 'Admin Integration'}, Object.assign({}, options, {columns: ['id']}))
```

Comment:
Had difficulty wrapping my head around why this is even needed here. After a little digging found that this piece is not just for when you are changing the key type, it also kicks in when the key is initially created (when creating an integration through the API for example). Don't see a good way of getting rid of this bit, unless we move this logic somehow to Ingeration creation phase (so that `role_id` is passed from there) which would be way cleaner :thinking:

Reply:
Yeah that's right! It's a little confusing I agree, we can probably clean it up a bit if we remove ability to change role/type and so we can explicitly just do it on "creation"